src/app/api/v1/user/welcome/route.ts

```diff
@@ -0,0 +1,102 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { getToken } from "next-auth/jwt";
+import { NextRequest, NextResponse } from "next/server";
+
+import { createProfile, isEligible } from "../../../../../external/onerep";
+import type { ProfileData } from "../../../../../external/onerep";
+import AppConstants from "../../../../../appConstants";
+// import { getL10n } from "../../../../functions/server/l10n";
+import {
+  getOnerepProfileId,
+  getSubscriberByEmail,
+} from "../../../../../db/tables/subscribers";
+
+import {
+  getLatestOnerepScan,
+  setOnerepProfileId,
+} from "../../../../../db/tables/onerep_scans";
+import { setProfileDetails } from "../../../../../db/tables/profiles";
+
+export async function POST(req: NextRequest) {
+  const token = await getToken({ req });
+
+  const eligible = await isEligible();
+  if (!eligible) {
+    throw new Error("User has no more manual scans left");
+  }
+
+  const params = new URLSearchParams(await req.text());
+
+  if (!params.has("firstname")) {
+    throw new Error("First name is required");
+  } else if (!params.has("lastname")) {
+    throw new Error("Last name is required");
+  } else if (!params.has("city")) {
+    throw new Error("City is required");
+  } else if (!params.has("state")) {
+    throw new Error("State is required");
+  } else if (!params.has("dob")) {
+    throw new Error("Date of Birth is required");
+  }
```

Comment:
Suggestion: Should we check not only the presence of the param but also the actual value so we are catching empty strings?

```suggestion
  const requiredParams = ['firstname', 'lastname', 'city', 'state', 'dob'];
  requiredParams.forEach(param => {
    if (!params.get(param)) {
      throw new Error(`${param} is required`);
    }
  });
```